packages/nx/src/utils/ignore.ts

```diff
@@ -0,0 +1,151 @@
+import * as fastGlob from 'fast-glob';
+import * as fs from 'fs';
+import ignore, { Ignore } from 'ignore';
+import * as path from 'path';
+import { Tree } from '../generators/tree';
+import { workspaceRoot } from './workspace-root';
+import { normalizePath } from './path';
+
+const workspaceIgnoreFiles = ['.gitignore', '.nxignore'];
+
+export function createWorkspaceIgnore(root = workspaceRoot): Ignore {
+  return createIgnoreFromFS(root, workspaceIgnoreFiles);
+}
+
+export function createNxignore(root: string = workspaceRoot): Ignore {
+  return createIgnoreFromFS(root, ['.nxignore']);
+}
+
+export function createGitignoreFromTree(tree: Tree): Ignore {
+  return createIgnoreFromTree(tree, ['.gitignore']);
+}
+
+export function readWorkspaceIgnorePatterns(root = workspaceRoot): string[] {
+  return combineIgnorePatterns(
+    workspaceIgnoreFiles.flatMap((ignoreFile) =>
+      readIgnoreFilesFromFS(root, ignoreFile)
+    )
+  );
+}
+
+export function createIgnoreFromPatterns(patterns: string[]): Ignore {
+  const ig = ignore();
+  ig.add(patterns);
+  return ig;
+}
+
+/**
+ * Reads ignore files from the file system and returns an object which can be
+ * used to check whether a file should be ignored.
+ *
+ * @param rootDir The directory in which to start searching for ignore files.
+ *                Paths evaluated by the returned object must be relative to this directory.
+ * @param ignoreFiles The filename of ignore files to include, e.g. ".gitignore"
+ */
+export function createIgnoreFromFS(
```

Comment:
I'm not sold on the name of this, but I'm not super sure on a replacement either. In general, we don't tend to have specifiers like "FromFS" on method names since that is the default mode of thinking. Perhaps just `createIgnore` would suffice? Alternatively, I actually prefer the `createWorkspaceIgnore` name, so this function could be renamed w/ the default params set.